app/src/main/java/com/serwylo/babyphone/editcontact/EditContactViewModel.kt

```diff
@@ -231,6 +239,107 @@ class EditContactViewModel(
         }
     }
 
+    fun shareContact(): Uri {
+        val fileName = contact.name.ifEmpty { contact.id.toString() }
+        val zipFile = File(context.getExternalFilesDir("Contacts"), "$fileName.zip")
```

Comment:
Do you mind if we change this to `"$fileName.babyphone.zip"`? The rationale is that in the future, I'd love to add an `<intent-filter>` to the app to listen for when people open such files in a file browser, or even downloading them from the internet. If it is just `.zip` with the contact name/id, then we would have to listen to all .zip files, which may provide a bad experience for people who want most `.zip` files to open in an archive manager.

Another way to future proof it would be to think about how one might allow multiple contacts to be put into a single zip file (again, imagine downloading a .zip file from online that contained 6 contacts). However, I don't think anything needs to change here. Rather, we could add a `.json` file in the zip file in the future which specifies which photos/sound files belong to each contact within the zip.

Having said that, while that is forwards-compatible, it is not backwards-compatible. One way to make this backwards-compatible would be to put the photo + sounds into a folder within the .zip file. When importing, open the first folder in the zip file.

Then, if in the future there is a zip file with multiple folders in it, old versions of the app will work just fine, and new versions will be able to import multiple contacts from the zip file. This would allow us to publish bundles of contacts online via GitHub pages in this repo or a sister repo, where people can download, I dunno, "Kittens and Puppies", or other quirky packages.

Reply:
I like your ideas! :+1: 
Especially exporting multiple contacts at once. Right now I have to share each contact individually with my partner.

I will change the file name to be compatible with a future intent-filter.